common/persistence/sql/sqlplugin/sqlite/plugin.go

```diff
@@ -0,0 +1,149 @@
+// The MIT License
+//
+// Copyright (c) 2021 Datadog, Inc.
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package sqlite
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	"github.com/jmoiron/sqlx"
+
+	"go.temporal.io/server/common/config"
+	"go.temporal.io/server/common/persistence/sql"
+	"go.temporal.io/server/common/persistence/sql/sqlplugin"
+	"go.temporal.io/server/common/resolver"
+)
+
+const (
+	// PluginName is the name of the plugin
+	PluginName = "sqlite3"
```

Comment:
Any thoughts on removing the version here? This would match the driver's `sqlite` package name and potentially help avoid confusion when managing config files.

I doubt we'd ever want to implement support for version 2 or earlier, and if a new major release of SQLite comes out we can always tack on a version number.

```suggestion
	PluginName = "sqlite"
```

Reply:
according to my memory the sqlx directly use plugin name as driver name?
ok to make the change

Reply:
This is decoupled in https://github.com/temporalio/temporal/blob/e8761cd0f11167e4cd25d03d7894569db2364a5d/common/persistence/sql/sqlplugin/sqlite/driver.go#L35 because the driver name differs between `github.com/mattn/go-sqlite3` and `godoc.org/modernc.org/sqlite`.